### DISCUSSION TITLE
In tauri, Use the official API interface, will the Rust API run faster than the JavaScript API?

### DISCUSSION THREAD

**Discussion by muyaCode** [tauri-apps/tauri#10365] [Q&A] [answered]
For example, which will make the program run faster using the Rust API than using the JavaScript API? Or are they both going at the same speed?

**Comment by FabianLars** [comment, marked as answer]
The JS Apis will run the Rust Apis behind the scenes but because of the communication between both sides, using the js Apis will be slower.

-> Using the Rust Apis is generally faster. If you need to return something to the js side from the rust code it's recommended to return only what's actually needed to keep the tax on the IPC low.

**Reply by muyaCode** [reply]
Thanks
